File: gateway/src/main/java/com/beat/gateway/shared/internal/config/RedisConfig.java
package com.beat.gateway.shared.internal.config;

import org.springframework.context.annotation.Configuration;
import org.springframework.data.redis.repository.configuration.EnableRedisRepositories;

import com.beat.gateway.guest.internal.store.GuestSessionRepository;
import com.beat.gateway.refreshtoken.internal.store.RefreshTokenRepository;

@Configuration(proxyBeanMethods = false)
@EnableRedisRepositories(basePackageClasses = {RefreshTokenRepository.class, GuestSessionRepository.class})
public class RedisConfig {
}

File: gateway/src/main/kotlin/com/beat/gateway/shared/internal/config/RedisConfig.kt
package com.beat.gateway.shared.internal.config

import com.beat.gateway.guest.internal.store.GuestSessionRepository
import com.beat.gateway.refreshtoken.internal.store.RefreshTokenRepository
import org.springframework.context.annotation.Configuration
import org.springframework.data.redis.repository.configuration.EnableRedisRepositories

@Configuration(proxyBeanMethods = false)
@EnableRedisRepositories(basePackageClasses = [RefreshTokenRepository::class, GuestSessionRepository::class])
class RedisConfig
